openmdao/components/external_code_comp.py

```diff
@@ -401,3 +401,16 @@ def apply_nonlinear(self, inputs, outputs, residuals):
             unscaled, dimensional residuals written to via residuals[key]
         """
         self._external_code_runner.run_component()
+
+    def solve_nonlinear(self, inputs, outputs):
```

Comment:
I think we need to make clear in the docs that, if the user overrides `apply_nonlinear` but does not override `solve_nonlinear` in their comp, then they are going to get a double run. This is different than with a regular ImplicitComponent, where if you don't define `solve_nonlinear`, then it does nothing, so it might cause some confusiion.

Reply:
i don't understand this comment. Plus, the user HAS to define apply_nonlinear?

Reply:
Since you know the subject way better than I do, can you give me the actual words to put into the doc?

Reply:
I posted in General to discuss further.